crates/router/src/core/relay.rs

```diff
@@ -0,0 +1,221 @@
+
+    let profile = db
+        .find_business_profile_by_merchant_id_profile_id(
+            key_manager_state,
+            &key_store,
+            merchant_id,
+            &profile_id_from_auth_layer,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::ProfileNotFound {
+            id: profile_id_from_auth_layer.get_string_repr().to_owned(),
+        })?;
+
+    let relay_response = match req.relay_type {
+        common_enums::RelayType::Refund => {
+            Box::pin(relay_refund(
+                state,
+                merchant_account,
+                profile,
+                key_store,
+                &req,
+            ))
+            .await?
+        }
+    };
+    Ok(hyperswitch_domain_models::api::ApplicationResponse::Json(
+        relay_response,
+    ))
+}
+
+pub async fn relay_refund(
+    state: SessionState,
+    merchant_account: domain::MerchantAccount,
+    profile: domain::Profile,
+    key_store: domain::MerchantKeyStore,
+    req: &relay_models::RelayRequest,
+) -> RouterResult<relay_models::RelayResponse> {
+    validate_relay_refund_request(req).attach_printable("Invalid relay refund request")?;
+
+    let db = state.store.as_ref();
+    let key_manager_state = &(&state).into();
+    let connector_id = &req.connector_id;
+
+    let merchant_id = merchant_account.get_id();
+
+    let connector_account = db
+        .find_by_merchant_connector_account_merchant_id_merchant_connector_id(
+            key_manager_state,
+            merchant_account.get_id(),
+            connector_id,
+            &key_store,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+            id: connector_id.get_string_repr().to_string(),
+        })?;
+
+    let connector_data = api::ConnectorData::get_connector_by_name(
+        &state.conf.connectors,
+        &connector_account.connector_name,
+        api::GetToken::Connector,
+        Some(connector_id.clone()),
+    )?;
+
+    let connector_integration: services::BoxedRefundConnectorIntegrationInterface<
+        api::Execute,
+        hyperswitch_domain_models::router_request_types::RefundsData,
+        hyperswitch_domain_models::router_response_types::RefundsResponseData,
+    > = connector_data.connector.get_connector_integration();
+
+    let relay_domain = get_relay_domain_model(req, merchant_account.get_id(), profile.get_id());
+
+    let relay_record = db
+        .insert_relay(key_manager_state, &key_store, relay_domain)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to insert a relay record in db")?;
+
+    let router_data = utils::construct_relay_refund_router_data(
+        &state,
+        &connector_account.connector_name,
+        merchant_id,
+        &connector_account,
+        &relay_record,
+    )
+    .await?;
+
+    let router_data_res = services::execute_connector_processing_step(
```

Comment:
Can we use execute_connector_processing_step with v2 router-data? If quickly it is possible means @hrithikesh026 . This change we can take in upcoming PR's